vision.md

```diff
@@ -0,0 +1,407 @@
+challenge and to focus on the business applications rather then connectivity and
+device management aspects, which are in a lot of cases considered "hygiene
+factors".
+thin-edge.io enables them to connect industrial sensors and automation to a cloud
+service.
+High assurances in quality and scalability of thin-edge.io, not only itself but
+also in combination with custom-developed components, let the Service Provider
+go from a POC implementation with thin-edge.io in their ecosystem to a rollout
+in production with little friction.
+
+**(Smart) Equipment makers/Hardware manufacturers/OEMs:** Equipment
+Manufacturers are moving away from focusing primarily on selling their equipment
+towards selling their equipment as a service (EaaS). An IIoT platform allowing
+to connect and manage assets as well as to use the visualize, analyze, and
+integrate equipment data is often the foundation to enable service-based
+business models and services. Here thin-edge.io is used as a foundation to bring
+"intelligence" around the equipment. Combined with any IoT platform, single
+purpose gateways or devices which sit on the equipment can be transformed into
+edge deployment options for services and applications that support the overall
+EaaS business model by leveraging thin-egdge.io framework.
+As equipment may be deployed remotely and not always physically accessible,
+thin-edge.io ensures a high level of robustness and security.
+
+**Smart equipment operators:** Rather than dealing with single equipment, smart
+operators are looking to enhance or optimize whole manufacturing processes with
+the help of IIoT. Operating complex manufacturing systems requires not only to
+handle various industry protocols and assets but also requires to connect and
+manage a heterogeneous set of industrial devices and hardware such as PLCs,
+protocol converters or industrial gateways. Here thin-edge.io helps to unify the
+complex device landscape by offering lightweight software modules which can run
+on resource constrained, brownfield hardware.
+The framework does so by providing a frictionless user experience to enable
+customers to develop their own solutions based on the included functionality but
+also by attaching custom functionality with minimum effort.
+
+## Users and Personas
+
+We define the following personas:
+
+- "End User"
+    - Person using a binary that has been built with the thin-edge.io framework
+      as well as a finished and working configuration
+
+- "System Integrator"
+    - Person using a binary that has been built with the thin-edge.io framework
+      and crafts a configuration suited to solve their problems
+
+- "Interface Developer"
+    - Person using the foreign interfaces of a thin-edge.io framework binary to
+      complement and extend its functionality.
+
+- "Plugin Developer"
+    - Person using the provided Rust crates to extend or improve binary plugins
+      written for the thin-edge.io framework
+
```

Comment:
```suggestion
An example for "Plugin Developer" is a **Device developer / Embedded engineer** whose background consists of Linux, C/C++, C#, embedded systems (IT focused):

- Responsible for
    - device logic including firmware and software
    - primarily only tasked to connect one or many devices/types to overall IoT
      solution
- Challenges and needs regarding device enablement:
    - enable new services and connectivity on the device while keeping stability
      and robustness (while having limited computing resources)
    - dealing with certificates, queuing, and persisting messages to handle
      unstable connections
    - allowing the device to be managed centrally, to keep it secure and
      up-to-date (while important to him not always #1 prio to overall
      initiative/project)

Also an "Embedded Developer" with OT focus is an example of a "Plugin Developer":

- familiar with PLCs, SCADA systems,
- dealing with the emerging need for connectivity and IIoT
- key concern is security, robustness, and resource efficiency which is
  usually overruling any "typical" IT solution on the device and implies
  some kind of custom logic. (e.g. rather closed OS, no dependencies can be
  installed, very strict certification and QA process, no CI/CD possible,
  long product lifecycle 10-20 years)
```

Reply:
These details don't help me. All the developers must be concerned by security and robustness. Their domain of expertise (business domain and development stack) might be diverse and independent of their use of thin-edge.

What is discriminant for us is their goal.
* An "agent developer" is focused on assembling components to build an application specific IoT agent (say the agents deployed by a smart meter company).
* A "plugin developer" is focused on providing access for a specific feature to any thin-edge software that might need it (e.g an Apama plugin).